.infrastructure/helm-chart/todoapp/templates/secret.yml

```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Chart.Name }}-secret
+  namespace: {{ .Values.todoapp.namespace }}
```

Comment:
Ensure that the `todoapp.namespace` value is defined in the `values.yaml` file. This is crucial for the correct deployment of the Secret in the specified namespace.